fix: add helm chart deployment and fix python package releases

dwmkerr opened the pull request:
- Add publish_helm_chart input option to deploy workflow
- Publish Helm chart to oci://{registry}/charts when enabled
- Enable Helm chart publishing in CICD automated releases
- Update build pipelines documentation
- Fixes the SDK not being released

<img width="928" height="356" alt="image" src="https://github.com/user-attachments/assets/1667ec86-3ee1-4f13-a348-0a71b0ea887e" />


<img width="654" height="582" alt="image" src="https://github.com/user-attachments/assets/7ddfe4bb-1151-4712-b7a7-4f9d24d21f24" />


<img width="619" height="546" alt="image" src="https://github.com/user-attachments/assets/52015733-c89a-4710-a1a9-5d15aa5edeef" />

<img width="1189" height="664" alt="image" src="https://github.com/user-attachments/assets/f37851cd-2bcd-4bae-915c-cc8fa62e5e89" />

dwmkerr committed:
feat: add optional Helm chart publishing to container registry

- Add publish_helm_chart input option to deploy workflow
- Publish Helm chart to oci://{registry}/charts when enabled
- Enable Helm chart publishing in CICD automated releases
- Update build pipelines documentation

dwmkerr committed:
fix: make publish-helm-chart independent job depending on check_version

- Extract Helm chart publishing from deploy-to-distribution-environment
- Add check_version dependency to deploy-to-pages for consistency

dwmkerr committed:
fix: remove checkout and hardcoded repo from helm chart deployment

- Remove unnecessary checkout step from deploy-helm-chart job
- Download chart from current repository instead of hardcoded McK-Internal/agents-at-scale
- Rename job and steps from publish to deploy for consistency

dwmkerr committed:
fix: add checkout step for gh release download

dwmkerr committed:
fix: rename publish_helm_chart to deploy_helm_chart for consistency

dwmkerr committed:
fix: add checkout and proper error handling to release-libs job

- Add missing checkout step for gh release upload to work
- Make job fail if wheel upload fails instead of silently continuing
- Ensures Python packages are properly attached to releases

skokaina commented on .github/workflows/deploy.yml:
```diff
@@ -239,7 +244,48 @@ jobs:
+        env:
+          GH_TOKEN: ${{ github.token }}
+
+      - name: Deploy Helm Chart to Container Registry
```
is there an interest to add some retry/timeout logic ? i.e using nick-fields/retry@v2

dwmkerr replied:
It's quite an effort to add an public github action to the allowed-list so I'd say it's something for later, also because this is to/from github releases and github CR the likely hood of failure is quite low (and our deploy pipeline is idempotent so we can re-run)

skokaina commented on .github/workflows/deploy.yml:
```diff
@@ -239,7 +244,48 @@ jobs:
+          gh release download "$VERSION" --pattern "ark-${CLEAN_VERSION}.tgz"
+          
+          echo "Downloaded chart:"
+          ls -la ark-${CLEAN_VERSION}.tgz
```
worth running a checksum check ?

dwmkerr replied:
again, NTH at this stage I'd like to at least get the chart out but I raised this: https://mckinsey.atlassian.net/browse/ARKQB-187

skokaina commented on .github/workflows/deploy.yml:
```diff
@@ -239,7 +244,48 @@ jobs:
+          CLEAN_VERSION="${{ needs.check_version.outputs.clean_version }}"
+          REGISTRY="${{ vars.DOCKER_REGISTRY || format('ghcr.io/{0}/{1}', github.repository_owner, github.event.repository.name) }}"
+          
+          helm registry login $REGISTRY -u "${{ secrets.DOCKER_REGISTRY_USERNAME || github.actor }}" --password-stdin <<< "${{ secrets.DOCKER_REGISTRY_PASSWORD || secrets.GITHUB_TOKEN }}"
```
do we sign the chart somewhere ? (i.e sigstore)

dwmkerr replied:
At the moment no, but again we can track this here: https://mckinsey.atlassian.net/browse/ARKQB-187

dwmkerr committed:
docs: remove architecture document from developer guide

dwmkerr committed:
docs: remove local development document from developer guide

dwmkerr committed:
docs: add ARK SDK installation guide

dwmkerr committed:
docs: update ARK SDK and deployment documentation

- Add uv command for ARK SDK installation
- Document Helm chart availability in releases and packages
- Show registry-based chart deployment as recommended approach

dwmkerr committed:
docs: remove stale docs and improve sdk/helm

dwmkerr committed:
Merge branch 'main' into build/deploy-helm-chart
